File: Data Structures & Algorithms/max-water-container/submission-0.go
func maxArea(heights []int) int {
    l,r := 0,len(heights)-1
    ans := -1
    for l < r{
        amount := min(heights[l],heights[r]) * (r-l)
        ans = max(ans,amount)
        if heights[l] < heights[r]{
            l++
        } else{
            r--
        }
    }
    return ans
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n =  s.size();
        array<int, 26> mp{};
        int l=0, r=0;
        int mx = 0;
        int ans = 0;

        while(r < n){
            mp[s[r] - 'A']++;
            mx = max(mx, mp[s[r] - 'A']);
            
            if((r-l+1)-mx > k){
                mp[s[l] - 'A']--;
                l++;
            }

            ans = max(ans, r-l+1);
            r++;
        }
        return ans;
    }
};
